### PR TITLE
ref(perf): Cleanup metric perf widget code

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
Removing some of the experimental UI we used to figure out metric-in-performance functionality.

- Will do a second pass later if there is extra cleanup

### CODE REVIEW

**Review by wmak — APPROVED** [review]
lgtm 🔥x4000

**Review by priscilawebdev — APPROVED** [review]

File: static/app/utils/metrics/metricsRequest.tsx [hunk 1]
```diff
@@ -1,249 +0,0 @@
-import * as React from 'react';
```
Comment (matejminar) [inline static/app/utils/metrics/metricsRequest.tsx:1]: Please leave this file.

File: tests/js/spec/utils/metrics/metricsRequest.spec.tsx [hunk 2]
```diff
@@ -1,285 +0,0 @@
-import {mountWithTheme, waitFor} from 'sentry-test/reactTestingLibrary';
```
Comment (matejminar) [inline tests/js/spec/utils/metrics/metricsRequest.spec.tsx:1]: and therefore this

**Review by matejminar — APPROVED** [review]
Looks good to me. I'll follow up with https://github.com/getsentry/sentry/pull/32483

**Comment by matejminar** [comment]
@k-fish I'll remove deprecated transform and remove it from MetricsRequest after this PR is merged

**Pull request merged**